sub-packages/bionemo-testing/src/bionemo/testing/megatron_parallel_state_utils.py

```diff
@@ -81,12 +84,24 @@ def _initialize_distributed_parallel_state(
     pipeline_model_parallel_split_rank: int = 0,
     context_parallel_size: int = 1,
     interactive: bool = False,
-) -> None:
+    precision: PrecisionTypes = "fp32",
+) -> pl.Trainer | None:
+    trainer = None
     # initialize pytorch DDP
     # if not interactive and not torch.distributed.is_initialized():
     if not torch.distributed.is_initialized():
-        logging.info("pytorch DDP is not initialized. Initializing with pytorch-lightening...")
-        trainer = pl.Trainer(devices=devices, strategy="ddp" if not interactive else "auto", num_nodes=1)
+        logging.info("pytorch DDP is not initialized. Initializing with pytorch-lightning...")
+        trainer = pl.Trainer(
+            devices=devices,
+            strategy="ddp" if not interactive else "auto",
+            num_nodes=1,
+            # plugins=nl.MegatronMixedPrecision(
+            #     precision=precision,
+            #     params_dtype=get_autocast_dtype(precision),
+            #     pipeline_dtype=get_autocast_dtype(precision),
+            #     autocast_enabled=False,
+            # ),
```

Comment:
@jstjohn, these lines aren't seemingly taking the place of `FP16Module` in `assert_model_equivalence`